IPython/terminal/shortcuts/__init__.py

```diff
@@ -181,30 +181,45 @@ def create_identifier(handler: Callable):
 ]
 
 AUTO_SUGGEST_BINDINGS = [
+    # there are two reasons for re-defining bindings defined upstream:
+    # 1) prompt-toolkit does not execute autosuggestion bindings in vi mode,
+    # 2) prompt-toolkit checks if we are at the end of text, not end of line
+    #    hence it does not work in multi-line mode of navigable provider
     Binding(
         auto_suggest.accept_in_vi_insert_mode,
         ["end"],
-        "default_buffer_focused & (ebivim | ~vi_insert_mode)",
+        "has_suggestion & default_buffer_focused & emacs_like_insert_mode",
```

Comment:
`(ebivim | ~vi_insert_mode)` appears incorrect as discussed in https://github.com/ipython/ipython/pull/13566#discussion_r1148561106